File: apps/mobile/src/api/authApi.ts
import { apiClient } from './apiClient';

export interface AuthUser {
  id: string;
  email: string;
  fullName: string;
  avatarUrl?: string;
}

export interface AuthResponse {
  user: AuthUser;
  token: string;
}

export const authApi = {
  async login(email: string, password: string): Promise<AuthResponse> {
    const { data } = await apiClient.post<AuthResponse>('/auth/login', { email, password });
    return data;
  },

  async register(email: string, password: string, fullName: string): Promise<AuthResponse> {
    const { data } = await apiClient.post<AuthResponse>('/auth/register', {
      email,
      password,
      fullName
    });
    return data;
  },

  async googleLogin(idToken: string): Promise<AuthResponse> {
    const { data } = await apiClient.post<AuthResponse>('/auth/google', { idToken });
    return data;
  },
};

File: apps/hub/app/src/main/kotlin/cl/babyguardian/hub/data/api/AuthApi.kt
package cl.babyguardian.hub.data.api

import cl.babyguardian.hub.data.model.GoogleLoginRequest
import cl.babyguardian.hub.data.model.LoginRequest
import cl.babyguardian.hub.data.model.LoginResponse
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthApi {
    @POST("auth/login")
    suspend fun login(@Body body: LoginRequest): LoginResponse

    @POST("auth/google")
    suspend fun googleLogin(@Body body: GoogleLoginRequest): LoginResponse
}
